### ISSUE TITLE
Module compiled with Swift 5.1 cannot be imported by the Swift 5.1.2 compiler

### ISSUE THREAD

**Issue by kenosan** [couchbase/couchbase-lite-ios#2570]
Upgraded to XCode 11.2 and received the compile error: 

> Module compiled with Swift 5.1 cannot be imported by the Swift 5.1.2 compiler: Pods/CouchbaseLite-Swift/iOS/CouchbaseLiteSwift.framework/Modules/CouchbaseLiteSwift.swiftmodule/x86_64-apple-ios-simulator.swiftmodule

This is the latest iteration on #2206 and #2225. There seems to be a solution to future proof this -- https://stackoverflow.com/a/58656323/19547

The SO answer mentions:
> You need to set the Build Libraries for Distribution option to Yes in your framework's build settings, otherwise the swift compiler doesn't generate the neccessary .swiftinterface files which are the key to future compilers being able to load your old library.
> 
> This ends up in your project.pbxproj file as:
> 
> BUILD_LIBRARY_FOR_DISTRIBUTION = YES;
> 
> After setting this flag, a framework I compiled using Xcode 11.0 (swift 5.1) was able to be compiled with Xcode 11.2 (swift 5.1.2) and everything appears to be working correctly.

Can we have this verified and updated for the project?

--------
* Version:   CouchbaseLite-Swift (2.6.1)
* Client OS: Xcode 11.2 with iOS 13

**Comment by meifu1709** [comment]
not work...

**Comment by meifu1709** [comment]
When can you solve this problem?

**Comment by rahimrahman** [comment]
While waiting for official release for this fix, I just build a new framework using instruction provided: https://github.com/couchbase/couchbase-lite-ios#how-to-build-the-framework-files

**Comment by jayahariv** [comment]
As @rahimrahman mentioned, you can built the CE version for now. We are working to get it out soon.

**Comment by fralways** [comment]
Will this be available soon? Already a lot of time passed.

**Comment by jayahariv** [comment]
CBL iOS 2.6.3 is available in Cocoapod and Carthage. Binary from the downloads page will appear soon.

**Comment by jayahariv** [comment]
Apologies, Cocoapod seems to be not published, but Carthage is.

**Comment by jayahariv** [comment]
Could you please try now, CBL iOS 2.6.3 for Cocoapods is available now.

**Comment by kenosan** [comment]
@jayahariv -- confirmed. 2.6.3 is available now in Cocoapods and got a successful compile and run 👍

**Comment by callo90** [comment]
Still not working with Carthage using 2.6.3 version downloading the binary.

**Comment by jayahariv** [comment]
@callo90 What error you are getting?

**Comment by callo90** [comment]
```Incompatible Swift version - framework was built with 5.1 (swiftlang-1100.0.270.13 clang-1100.0.33.7) and the local version is 5.1.2 (swiftlang-1100.0.278 clang-1100.0.33.9).``` @jayahariv

**Comment by jayahariv** [comment]
HI @callo90, this is a known issue from Carthage, even though the library is compatible, Carthage is checking for local swift compiler version with compiler used to built the binary. https://github.com/Carthage/Carthage/issues/2887

**Comment by jayahariv** [comment]
@callo90  As a workaround, you could switch the xcode when you update and then change it back. Framework downloaded will be module compatible.
```
# revert to Xcode 11.1 
sudo xcode-select -switch /Applications/Xcode11.1.app/

carthage update

# switch back to newer xcode
sudo xcode-select -switch /Applications/Xcode-beta.app/
```

<img width="1277" alt="Screen Shot 2019-11-26 at 10 13 27 AM" src="https://user-images.githubusercontent.com/10448770/69660607-8641ed00-1035-11ea-8686-b2f4ebc30117.png">

**Comment by sam-w** [comment]
For anyone watching this issue, the problem should now be fixed with the release of Carthage 0.35.0
